lib/scss-variables-compiler.js

```diff
@@ -13,14 +14,28 @@ function makeBreakpointLine(bp, isLast) {
   return `  ${bp.name}: ('${bp.prefix}', ${bp.begin})${isLast ? '' : ','}`;
 }
 
-module.exports = function compile(base, config) {
-  let filePath = PATH.join(base, '_variables.scss');
+module.exports = class FlexiVariableCompiler extends Plugin {
+  constructor(node, options) {
+    super([node], {
+      persistentOutput: true
+    });
+    this.options = options;
 
-  // clean out the old _variables.scss
-  if (FS.existsSync(filePath)) {
-    FS.unlinkSync(filePath);
+    this._hasBuilt = false;
   }
 
+  build() {
+    if (this._hasBuilt) { return; }
```

Comment:
@stefanpenner For my own education.. how might we end up calling `build()` a second time after the first call to `build()`? Also, much thanks for this PR, it has been very educational on broccoli as is!